cpu/stm32/periph/adc_l4.c

```diff
@@ -28,10 +28,22 @@
 #include "periph/vbat.h"
 #include "ztimer.h"
 
+/**
+ * @brief Not all STM32 L4 boards have 3 ADC devices
+ * for example, L4R5ZI has only one ADC
+ */
+#if defined ADC_DEVS && ADC_DEVS == 1
+#define ADC	ADC1_COMMON
+#endif
+#if defined ADC_DEVS && ADC_DEVS == 3
+#define ADC	ADC123_COMMON
```

Comment:
I'm a bit confuses - if there is only one ADC device, shouldn't `ADC_DEVS` be set to 1?

Reply:
This particular L4 board has one ADC but other L4 boards have 3 devices. Number of devices - `ADC_DEVS` is set in other file ([cpu/stm32/include/periph/l4/periph_cpu.h](https://github.com/RIOT-OS/RIOT/pull/18676/files#diff-1b6703911b423a46fd47ff6d81f395d47ff69a8390e8471c1f9923ae40092ae9)) depending on the particular microcontroller.

The previous code used directly `ADC123_COMMON` define from CMSIS - which did nit work for STM32L4R5 so I added this if and new define - `ADC`.

I committed other white space and empty line fixes.